lib/ASTGen/Sources/ASTGen/DeclAttrs.swift

```diff
@@ -1445,6 +1445,141 @@ extension ASTGenVisitor {
     )
   }
 
+  func generateValueOrType(expr node: ExprSyntax) -> BridgedTypeRepr? {
+    var node = node
+
+    // Try value first.
+    let minusLoc: BridgedSourceLoc
+    if let prefixExpr = node.as(PrefixOperatorExprSyntax.self),
+      prefixExpr.operator.rawText == "-",
+      prefixExpr.expression.is(IntegerLiteralExprSyntax.self) {
+      minusLoc = self.generateSourceLoc(prefixExpr.operator)
```

Comment:
Should it be an error/assertion if there's a `minusLoc` without an integer literal?

Reply:
In such cases, it falls back to `generateTypeRepr` which should diagnose it as an invalid type.

Reply:
Hmm, not if we do `node = prefixExpr.expression`, right?

Reply:
oooh, indeed. thanks!

Reply:
Oh wait there's a check `prefixExpr.expression.is(IntegerLiteralExprSyntax.self)` right before this line.
Though, I will add an assertion `minusLoc == nil` before the `generateTypeRepr(_:)` call.

Reply:
Oh hah, sorry missed that